client/lib/plugins/test/server-dom-parser.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import {
+	sanitizeSectionContent as clean,
+	overrideSanitizeSectionRoot,
+	getSanitizeSectionRoot,
+} from '../sanitize-section-content';
+import { getServerRoot } from '../server-dom-parser';
+
+const root = getServerRoot();
+overrideSanitizeSectionRoot( root );
+/**
+ * Attempts to create a DOM node from given HTML
+ *
+ * @param {string} html expected HTML to create node
+ * @returns {object | null} possible node described by HTML
+ */
+const cleanNode = ( html ) => {
+	const div = document.createElement( 'div' );
+
+	div.innerHTML = clean( html );
+
+	return div.firstChild;
+};
+
+test( 'should change the sanitization root object', () => {
+	expect( root ).toBe( getSanitizeSectionRoot() );
+} );
+test( 'should not strip allowed tags', () =>
+	expect( clean( '<div>👍</div>' ) ).toBe( '<div>👍</div>' ) );
+
+test( 'should strip out disallowed tags', () =>
+	expect( clean( '<marquee></marquee>' ) ).toBe( '' ) );
+
+test( 'should preserve children of stripped tags', () =>
+	expect( clean( '<unsupported><b>👍</b></unsupported>' ) ).toBe( '<b>👍</b>' ) );
+
+test( 'should strip out content with disallowed tags', () =>
```

Comment:
This test made me wonder: where are the disallowed tags and attributes defined? Any way we can get it work as close as possible to what `wp_kses()` does?

Reply:
Good point! they were defined through a couple of switches ([here](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/plugins/sanitize-section-content.js#L16) and [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/plugins/sanitize-section-content.js#L57)).
I replaced them with something similar to the `wp_kses()` allowed tags configurations.